src/librustc_middle/query/mod.rs

```diff
@@ -669,29 +669,22 @@ rustc_queries! {
     }
 
     Other {
-        /// Evaluates a constant without running sanity checks.
-        ///
-        /// **Do not use this** outside const eval. Const eval uses this to break query cycles
-        /// during validation. Please add a comment to every use site explaining why using
-        /// `const_eval_validated` isn't sufficient. The returned constant also isn't in a suitable
-        /// form to be used outside of const eval.
-        query const_eval_raw(key: ty::ParamEnvAnd<'tcx, GlobalId<'tcx>>)
+        /// Evaluates a constant and returns the computed allocation.
+        query const_eval(key: ty::ParamEnvAnd<'tcx, GlobalId<'tcx>>)
             -> ConstEvalRawResult<'tcx> {
```

Comment:
We should make a note to change the type of this type -- and its doc comment also needs adjusting.

Reply:
this isn't the API anyone is using, it's just the query behind all the actual invocation functions

Reply:
Sure. It's still a confusing type name and its doc comment (i.e., the doc comment of `ConstEvalRawResult`) still talks about the "raw" query.